data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/CsaProfileCracCreatorTest.java

```diff
@@ -0,0 +1,246 @@
+        assertEquals("RTE_CO5", c11.getName());
+        assertEquals(1, c11.getNetworkElements().size());
+        Iterator<NetworkElement> it11 = c11.getNetworkElements().iterator();
+        assertEquals("f0dee14e-aa43-411e-a2ea-b9879c20f3be", it11.next().getId());
+
+        Contingency c12 = it.next();
+        assertEquals("7e31c67d-67ba-4592-8ac1-9e806d697c8e", c12.getId());
+        assertEquals("ELIA_CO2", c12.getName());
+        assertEquals(1, c12.getNetworkElements().size());
+        Iterator<NetworkElement> it12 = c12.getNetworkElements().iterator();
+        assertEquals("536f4b84-db4c-4545-96e9-bb5a87f65d13 + d9622e7f-5bf0-4e7e-b766-b8596c6fe4ae", it12.next().getId());
+
+        Contingency c13 = it.next();
+        assertEquals("37997e71-cb7d-4a8c-baa6-2a1594956da9", c13.getId());
+        assertEquals("ELIA_CO3", c13.getName());
+        assertEquals(1, c13.getNetworkElements().size());
+        Iterator<NetworkElement> it13 = c13.getNetworkElements().iterator();
+        assertEquals("550ebe0d-f2b2-48c1-991f-cebea43a21aa", it13.next().getId());
+
+        Contingency c14 = it.next();
+        assertEquals("ce19dd34-429e-4b72-8813-7615cc57b4a4", c14.getId());
+        assertEquals("RTE_CO6", c14.getName());
+        assertEquals(1, c14.getNetworkElements().size());
+        Iterator<NetworkElement> it14 = c14.getNetworkElements().iterator();
+        assertEquals("04839777-c766-11e1-8775-005056c00008", it14.next().getId());
+
+        Contingency c15 = it.next();
+        assertEquals("475ba18f-cbf5-490b-b65d-e8e03f9bcbc4", c15.getId());
+        assertEquals("RTE_CO2", c15.getName());
+        assertEquals(1, c15.getNetworkElements().size());
+        Iterator<NetworkElement> it15 = c15.getNetworkElements().iterator();
+        assertEquals("e02e1166-1c43-4a4d-8c5a-82298ee0c8f5", it15.next().getId());
+    }
+
+    @Test
+    public void testCreateCracCSATestWithRejectedFiles() {
+        CsaProfileCracImporter cracImporter = new CsaProfileCracImporter();
+        InputStream inputStream = getClass().getResourceAsStream("/CSA_Test_With_Rejected_Files.zip");
+        CsaProfileCrac nativeCrac = cracImporter.importNativeCrac(inputStream);
+
+        Properties importParams = new Properties();
+        Network network = Network.read(Paths.get(new File(CsaProfileCracCreatorTest.class.getResource("/CSA_Test_With_Rejected_Files.zip").getFile()).toString()), LocalComputationManager.getDefault(), Suppliers.memoize(ImportConfig::load).get(), importParams);
+
+        CsaProfileCracCreator cracCreator = new CsaProfileCracCreator();
+        CsaProfileCracCreationContext cracCreationContext = cracCreator.createCrac(nativeCrac, network, OffsetDateTime.parse("2023-03-29T12:00Z"), new CracCreationParameters());
+
+        assertNotNull(cracCreationContext);
+        assertTrue(cracCreationContext.isCreationSuccessful());
+        assertEquals(7, cracCreationContext.getCrac().getContingencies().size());
```

Comment:
most of the contingencies should have been refused for some reason. The CRAC in this case is expected to contain only 1 or 2 contingencies from ELIA

Reply:
I add an unit test for the last US use case